Subject: [PATCH] Fixed NPE crash on Dialpad

The crash occurs because Context instance already detached is used.
Fixed to check if getActivity() is NULL.

diff --git a/src/com/android/dialer/dialpad/DialpadFragment.java b/src/com/android/dialer/dialpad/DialpadFragment.java
index 55d5346..07bce0e 100644
--- a/src/com/android/dialer/dialpad/DialpadFragment.java
+++ b/src/com/android/dialer/dialpad/DialpadFragment.java
@@ -360,11 +360,15 @@
 
     @Override
     public void afterTextChanged(Editable input) {
+        final Activity activity = getActivity();
+        if (activity == null) {
+            return;
+        }
         // When DTMF dialpad buttons are being pressed, we delay SpecialCharSequenceMgr sequence,
         // since some of SpecialCharSequenceMgr's behavior is too abrupt for the "touch-down"
         // behavior.
         if (!mDigitsFilledByIntent &&
-                SpecialCharSequenceMgr.handleChars(getActivity(), input.toString(), mDigits)) {
+                SpecialCharSequenceMgr.handleChars(activity, input.toString(), mDigits)) {
             // A special sequence was entered, clear the digits
             mDigits.getText().clear();
         }
